### PR TITLE
more arc features

### PR DESCRIPTION
**Pull request by Araq**

### CODE REVIEW

File: lib/system/repr_v2.nim [hunk 1]
```diff
@@ -0,0 +1,170 @@
...
+  ## converted to the string "false" or "true".
+
+proc repr*(x: char): string {.magic: "CharToStr", noSideEffect.}
+  ## repr for a character argument. Returns `x`
+  ## converted to a string.
+  ##
+  ## .. code-block:: Nim
+  ##   assert $'c' == "c"
+
+proc repr*(x: cstring): string {.magic: "CStrToStr", noSideEffect.}
```
Comment (timotheecour) [inline lib/system/repr_v2.nim:24]: but that just gives `$x` unlike standard repr ; IIRC repr was supposed to you you more low level debug info (eg the pointers)
```nim
proc repr2*(x: cstring): string {.magic: "CStrToStr", noSideEffect.}
var x = "abc"
var x2 = x.cstring
echo x.repr
echo x2.repr
echo x.repr2
echo x2.repr2
```
shows:
```
0x1082a0058"abc"
0x1082a0058"abc"

abc
abc
```

File: lib/system/repr_v2.nim [hunk 2]
```diff
@@ -0,0 +1,170 @@
...
+  ## repr for an enumeration argument. This works for
+  ## any enumeration type thanks to compiler magic.
+  ##
+  ## If a `repr` operator for a concrete enumeration is provided, this is
+  ## used instead. (In other words: *Overwriting* is possible.)
+
+template repr(t: typedesc): string = $t
+
+proc isNamedTuple(T: typedesc): bool =
+  # Taken from typetraits.
```
Comment (timotheecour) [inline lib/system/repr_v2.nim:43]: old isNamedTuple had issues, need to use new isNamedTuple magic

**Comment by Araq** [comment]
Will address your remarks in a followup PR.

**Comment by timotheecour** [comment]
> Will address your remarks in a followup PR.

cool (maybe for this it's good to add top level doc comment `experimental module` to indicate api usage or semantics (cf `repr`) can change)

File: lib/system/repr_v2.nim (hunk 2 above)
Comment (timotheecour) [reply lib/system/repr_v2.nim:43]: got fixed in #13268

**Pull request merged**